### ISSUE TITLE
[Enhancement] Consider externalizing the rules for runtime construction

### ISSUE THREAD

**Issue by grgcombs** [netguru/ngrvalidator#5]
Would you consider mechanisms that (optionally) externalize the validation rules?  For "design-time-at-runtime" situations, we don't know the properties of the data models until the app is installed on the device and the user logs in to their backend server.

We're experimenting with JSON-Schema and I think that might be useful here...
- [KiteJSONValidator](https://github.com/samskiter/KiteJSONValidator)
- [JSONTools](https://github.com/grgcombs/JSONTools) _(Pardon the self-advertisement)_

**Comment by PatrykKaczmarek** [comment]
Hello @grgcombs 

First of all sorry for late response. I was very busy lately. 

I also considered about functionality you proposed. It would be neat feature but unfortunately out of `NGRValidator` scope. The main purpose of `NGRValidator` is to validate user inputs and/or data which are going outside of an app rather than validating JSON/XML schema. There are some open source repositories (you mentioned about 2 of them) which do that job. I think it is quite extensive topic for separate pod.

To sum up `NGRValidator` at **least now and in the near future** will not have JSON schema validation.
